### PR TITLE
Speed up end_to_end tests

### PR DESCRIPTION
**Pull request by chrishajas**
Instead of dropping and recreating the database on every scenario, drop and
recreate the schema. This reduces the total test time from ~70s to ~40s.
Each individual scenario now takes ~2-3s.

Author: Chris Hajas <chajas@pivotal.io>

### CODE REVIEW

File: end_to_end/end_to_end_suite_test.go [hunk 1]
```diff
@@ -83,13 +87,12 @@ var _ = Describe("backup end to end integration tests", func() {
 			if restoreConn != nil {
 				restoreConn.Close()
 			}
-			exec.Command("dropdb", "restoredb").Run()
-			err := exec.Command("createdb", "restoredb").Run()
+			restoreConn = utils.NewDBConn("restoredb")
```
Comment (khuddlefish) [inline end_to_end/end_to_end_suite_test.go]: If we don't drop and restore the db every time, the DBConn creation should probably go in BeforeSuite

File: end_to_end/end_to_end_suite_test.go [hunk 2]
```diff
@@ -197,8 +200,7 @@ var _ = Describe("backup end to end integration tests", func() {
 		})
 		It("runs gpbackup and gprestore on database with all objects", func() {
 			backupConn.Close()
```
Comment (khuddlefish) [inline end_to_end/end_to_end_suite_test.go]: Don't need to close the connection if we're not dropping the database

File: end_to_end/end_to_end_suite_test.go [hunk 3]
```diff
@@ -197,8 +200,7 @@ var _ = Describe("backup end to end integration tests", func() {
...
-			exec.Command("dropdb", "testdb").Run()
-			err := exec.Command("createdb", "testdb").Run()
+			err := exec.Command("psql", "-d", "testdb", "-c", "DROP SCHEMA IF EXISTS schema2 CASCADE; DROP SCHEMA public CASCADE; CREATE SCHEMA public;").Run()
```
Comment (khuddlefish) [inline end_to_end/end_to_end_suite_test.go]: We could probably use testutils.AssertQueryRuns for these statements. Would be a bit more consistent with the rest of our tests.

File: end_to_end/end_to_end_suite_test.go (hunk 2 above)
Comment (khuddlefish) [reply end_to_end/end_to_end_suite_test.go]: Can also take out creating the new backupConn at the end of this test

File: end_to_end/end_to_end_suite_test.go [hunk 4]
```diff
@@ -65,6 +65,10 @@ var _ = Describe("backup end to end integration tests", func() {
 		if err != nil {
 			Fail(fmt.Sprintf("%v", err))
 		}
+		err = exec.Command("createdb", "restoredb").Run()
+		if err != nil {
+			Fail(fmt.Sprintf("%v", err))
+		}
 		backupConn = utils.NewDBConn("testdb")
 		backupConn.Connect()
 		utils.ExecuteSQLFile(backupConn, "test_tables.sql")
```
Comment (khuddlefish) [inline end_to_end/end_to_end_suite_test.go]: We could also change this to be like 
``` 
connStr := []string{
"-d", "testdb",
"-f", "all_objects.sql",
"-q",
}
exec.Command("psql", connStr...)
```
below and then we can remove the `ExecuteSQLFile` function.

Comment (chrishajas) [reply end_to_end/end_to_end_suite_test.go]: The ExecuteSQLFile function also sets `ON_ERROR_STOP=1`, which prevents the swallowing of errors. I think instead we should use this function when executing sql files so we don't accidentally commit bad syntax.

File: end_to_end/end_to_end_suite_test.go (hunk 1 above)
Comment (chrishajas) [reply end_to_end/end_to_end_suite_test.go]: Fixed, thanks!

File: end_to_end/end_to_end_suite_test.go (hunk 2 above)
Comment (chrishajas) [reply end_to_end/end_to_end_suite_test.go]: Fixed, thanks!

File: end_to_end/end_to_end_suite_test.go (hunk 3 above)
Comment (chrishajas) [reply end_to_end/end_to_end_suite_test.go]: Fixed, thanks!

**Comment by khuddlefish** [comment]
Pushed to master!

**Pull request closed without merging**